packages/backend/src/open_payments/payment/outgoing/service.ts

```diff
@@ -344,6 +341,25 @@ async function fundPayment(
       return FundingError.WrongState
     }
     if (amount !== payment.sendAmount.value) return FundingError.InvalidAmount
+
+    // Create the outgoing payment liquidity account before trying to transfer funds to it.
+    try {
+      await deps.accountingService.createLiquidityAccount({
+        id: id,
+        asset: payment.asset
+      })
+    } catch (err) {
+      // Don't complain if liquidity account already exists.
+      if (
+        err instanceof CreateAccountError &&
+        err.code === CreateAccountErrorCode.exists
+      ) {
+        // Do nothing.
+      } else {
+        throw err
+      }
```

Comment:
Similar to my other comment: Ignore `exists_with_different_{debits,credits}_{pending,posted}` here. If this SQL transaction aborts part-way through and restarts, the TB account may already exist (from the previous `accountingService.createLiquidityAccount`) with money (from the previous `accountingService.createDeposit`) without the `state` being updated to `Sending`.